monai/transforms/rand_rotation.py

```diff
@@ -0,0 +1,210 @@
+        """
+        raise NotImplementedError
+
+
+class Transform(object):
+    """
+    An abstract class of a ``Transform``.
+    A transform is callable that processes ``data``.
+
+    It could be stateful and modify ``data`` in place.
+    Therefore the implementation should be aware of:
+    - thread-safety when mutating its own states.
+        When used from a multi-process context, this class's states are read-only.
+    - ``data`` content unused by this transform may still be used in the
+        subsequent transforms in a composed transform.
+        see also: `monai.transforms.compose.Compose`.
+    - storing too much information in ``data`` may not scale.
+    """
+
+    def __call__(self, data):
+        """
+        ``data`` is an element which often comes from an iteration over an
+        iterable, such as``torch.utils.data.Dataset``. This method should
+        return an updated version of ``data``.
+        """
+        raise NotImplementedError
+
+
+class MapTransform(Transform):
+    """
+    A subclass of ``Transform`` with an assumption that the ``data`` input of
+    ``self.__call__`` is a MutableMapping such as ``dict``.
+
+    The ``keys`` parameter will be used to get and set the actual data
+    item to transform.  That is, the callable of this transform should
+    follow the pattern:
+    ```
+        def __call__(self, data):
+            for key in self.keys:
+                if key in data:
+                    update output data with some_transform_function(data[key]).
+                else:
+                    do nothing or some exceptions handling.
+            return data
+    ```
+    """
+
+    def __init__(self, keys):
+        self.keys = keys if isinstance(keys, (list, tuple)) else (keys,)
+        if not self.keys:
+            raise ValueError('keys unspecified')
+        for key in self.keys:
+            if not isinstance(key, Hashable):
+                raise ValueError('keys should be a hashable or a list of hashables, got {}'.format(type(key)))
+
+
+@export
+class Rotate90(Transform):
+    """
+    Rotate an array by 90 degrees in the plane specified by `axes`.
+    """
+
+    def __init__(self, k=1, axes=(1, 2)):
+        """
+        Args:
+            k (int): number of times to rotate by 90 degrees.
+            axes (2 ints): defines the plane to rotate with 2 axes.
+        """
+        self.k = k
+        self.plane_axes = axes
+
+    def __call__(self, img):
+        return np.rot90(img, self.k, self.plane_axes)
+
+
+@export
+class Rotate90d(MapTransform):
+    """
+    dictionary-based wrapper of Rotate90.
+    """
+
+    def __init__(self, keys, k=1, axes=(1, 2)):
+        """
+        Args:
+            k (int): number of times to rotate by 90 degrees.
+            axes (2 ints): defines the plane to rotate with 2 axes.
+        """
+        MapTransform.__init__(self, keys)
+        self.k = k
+        self.plane_axes = axes
+
+    def __call__(self, data):
+        d = copy(data)
```

Comment:
If `data` is a dictionary, `d = dict(data)` would also work if you wanted to reduce the number of imports.